packages/trees/frontend/sirius-components-trees/src/views/useExpandAll.tsx

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useLazyQuery } from '@apollo/client';
+import { GQLGetExpandAllTreePathData, GQLGetExpandAllTreePathVariables } from './TreeView.types';
+import { useEffect, useState } from 'react';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { UseExpandAllState, UseExpandAllValue } from './useExpandAll.types';
+
+const getExpandAllTreePathQuery = gql`
+  query getExpandAllTreePath($editingContextId: ID!, $treeId: ID!, $treeItemId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        expandAllTreePath(treeId: $treeId, treeItemId: $treeItemId) {
+          treeItemIdsToExpand
+          maxDepth
+        }
+      }
+    }
+  }
+`;
+
+export const useExpandAll = (expanded: string[], maxDepth: number): UseExpandAllValue => {
```

Comment:
I created a hook for the "default expand all behavior", because this feature is not only needed by the _ExpandAll_ tool in the contextual menu, but also by the model browser and selection widget.
The new Sirius Web _ExpandAll_ contribution uses this hook, but downstream applications can override it while keeping the existing behavior of the model browser and reference widget.

Reply:
Since this hook does not perform the `expandAll`, it should probably be named `useExpandAllTreePath`

Reply:
done